addons/mail/static/src/components/composer/composer.scss

```diff
@@ -231,8 +233,12 @@
 
 .o_Composer_coreMain:not(.o-composer-is-compact) {
     background: white;
-    border-radius: 5px;
     border: 1px solid lighten(gray('400'), 5%);
+
+    // textarea should not be rounded when there is subject field above
```

Comment:
Phrase is actually backwards, as the rule is not removing an existing rounded, but adding it in some cases. Proper way is:
texteara should be rounded but only when there is no subject field above